spec/services/mixtape_maker_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+describe MixtapeMaker do
+  describe '.generate' do
+    it 'makes a correctly formatted mixtape' do
+      artists = create_list(:artist, 3)
+      songs = create_list(:song, 4, artist: artists[0])
+      mixtape = MixtapeMaker.generate(2, artist_ids: artists.collect{ |obj| obj.id})
```

Comment:
Have a look at the `pluck(:column)` method.

http://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-pluck

That's a more common way to extract an attribute (or subset of attributes) from a collection for ActiveRecord items.

The difference is that pluck will do a SQL `select` vs collect or map will iterate the collection and get the value.

It's usually faster to let the db do the work -- especially if the objects aren't eager loaded as pluck does 1 database call whereas collect/map could do n+1.